src/components/menu/__internal__/keyboard-navigation/index.js

```diff
@@ -7,6 +7,10 @@ function characterNavigation(event, focusableItems, currentFocusedIndex) {
   let nextMatch;
   const selectedKey = event.key.toLowerCase();
   const getMenuText = (element) => {
+    if (element.keyboardOverride) {
```

Comment:
I think we need a unit test specifically testing this, i.e. character navigating to the MenuItem with keyboardOverride